### ISSUE TITLE
Update replication system

### ISSUE THREAD

**Issue by thedodd** [hadron-project/hadron#57]
The scheduler assignments system will be landing soon. Once that is done, stream & pipeline controllers can respond to assignment info to drive their logic.

Controllers will need to be aware of their status as leader, and when they are a partition leader, they will need to replicate data to the replicas of their partition.

**Comment by thedodd** [comment]
Superseded by RFC 008.
